src/multi_record_log.rs

```diff
@@ -82,25 +83,35 @@ impl MultiRecordLog {
         directory_path: &Path,
         sync_policy: SyncPolicy,
     ) -> Result<Self, ReadRecordError> {
+        // io errors are non-recoverable
         let rolling_reader = crate::rolling::RollingReader::open(directory_path).await?;
         let mut record_reader = crate::recordlog::RecordReader::open(rolling_reader);
         let mut in_mem_queues = crate::mem::MemQueues::default();
         loop {
             let file_number = record_reader.read().current_file().clone();
-            if let Some(record) = record_reader.read_record().await? {
+            let Ok(record) = record_reader.read_record().await else {
+                warn!("Detected corrupted record: some data may have been lost");
+                continue;
+            };
+            if let Some(record) = record {
                 match record {
                     MultiPlexedRecord::AppendRecords {
                         queue,
                         records,
                         position,
                     } => {
                         if !in_mem_queues.contains_queue(queue) {
-                            in_mem_queues
-                                .ack_position(queue, position)
-                                .map_err(|_| ReadRecordError::Corruption)?;
+                            in_mem_queues.ack_position(queue, position);
                         }
                         for record in records {
+                            // if this fails, it means some corruption wasn't detected at a lower
+                            // level, or we wrote invalid data.
                             let (position, payload) = record?;
+                            // this can fail if queue doesn't exist (it was created just above, so
```

Comment:
Should we autocreate queue for regular append operation? 
I do not mean when reading the record log.

I mean the two following scenario:
- create queue
- append record
- corrupt touch queue record
- restart
=> outcome : the queue is there with the record.

VS

- create queue
- corrupt touch queue record
- restart
- append record
=> outcome : the append record operation is wrong.

Case for the current implementation:
In the first case, accepting the append record is justified:
its presence in the record log is proof enough that we accepted the queue in the past.
Also if the main of corruption is crashes.

Case for accepting auto creating queues on append if it does not exist.
It is simple to reason with...

Reply:
I was wondering the same, so I searched in QW source code (as it's the main consumer of mrecordlog), and it seems QW will create a queue if it doesn't exist when instantiating an `IngestApiSource`.
On the other hand, I see a slight danger in creating a queue like that. If by mistake you push on a queue which shouldn't exist, and forget to truncate it, the ondisk files can't be gced

Reply:
> the ondisk files can't be gced

Excellent point!